### PR TITLE
Error Infrastructure Enhancement

### PR DESCRIPTION
**Pull request by miloszm**
Error Infrastructure Enhancement

Implements #132

### CODE REVIEW

File: piecrust/Cargo.toml [hunk 1]
```diff
@@ -19,6 +19,7 @@ colored = "2"
 libc = { version = "^0.2", default-features = false }
 region = "3"
 tempfile = "3"
+thiserror = "1.0"
```
Comment (ureeves) [inline piecrust/Cargo.toml:22]: Since the others above only specify major version, perhaps we should do the same with this one.

Comment (ureeves) [reply piecrust/Cargo.toml:22]:
```suggestion
thiserror = "1"
```

**Review by ureeves — APPROVED** [review]
One nit, but otherwise LGTM

**Review by herr-seppia — APPROVED** [review]
LGTM

**Pull request merged**